core/src/components/checkbox/checkbox.ionic.scss

```diff
@@ -25,4 +27,20 @@
 
    // Size
    --size: #{$checkbox-ionic-small-size};
-}
\ No newline at end of file
+}
+
+// Ionic Design Checkbox Required
+// --------------------------------------------------
+:host(.ion-invalid) {
+  .checkbox-icon {
+    border-color: #f72c2c;
+  }
+}
+
+// Focused
+:host(.ion-focused.ion-invalid) .native-wrapper {
```

Comment:
These styles build on the changes you have here: https://github.com/ionic-team/ionic-framework/pull/29286/files#diff-796ffc72b6171910c25d5aad085bc245095c38f104d55e253e06282702205066R77-R81.

If we wait for that PR to be reviewed first, this implementation can be simplified to:

```scss
:host(.ion-invalid) {
  --focus-ring-color: #ffafaf;
}
```